### PR TITLE
course-outline

### PR DESCRIPTION
**Pull request by BonnieJRobert**
Please review README.md and course-outline-md.  

To view the html version without rendering, prefix with the GitHub url with https://htmlpreview.github.io/? like this:
https://htmlpreview.github.io/?https://github.com/bcgov/ds-intro-to-git/blob/learning-objectives/course-outline.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,61 @@
 Introduction To Version Control With Git and GitHub
 ============================
 
+This is a repository to house materials for a 1/2 day workshop introducing participants to version control using git and the popular code hosting platform, GitHub.
+
+The goal of this workshop is to teach new-to-programming data professionals how to harness the power of version control by giving attendees a foundation in the fundamentals of version control and git.
```
Comment (lindsay-fredrick) [inline README.md:8]: Not necessarily new to programming but likely newer to version control?

File: README.md [hunk 2]
```diff
@@ -3,6 +3,61 @@
...
+
+GitHub is a widely-used platform for hosting and sharing code repositories, facilitating seamless collaboration among developers.  Git, on the other hand, is a popular version control system that enables you to track changes in your code, collaborate with team members, and maintain a history of your project's development. 
+
+By the end of this course, you will have gained a fundamental understanding of version control, enabling you to:
+
+- Set up a github account and install git (pre-workshop)
+- Understand the value-add of using version control to manage software projects
+- Understand the basics of how automated version control systems work.
+- Use command-line git to track local file changes
+- (*more about git*)
```
Comment (lindsay-fredrick) [inline README.md:18]: Don't think we need anything more here, high level overview should suffice.

File: README.md [hunk 3]
```diff
@@ -3,6 +3,61 @@
...
+- What are remote hosting platforms and why do we use them?
+- Connect to GitHub, create a remote repository and push local changes to GitHub
+- (*Something about BC Gov specific materials Org, Licensing, bcgovr*)
```
Comment (lindsay-fredrick) [inline README.md:21]:
```suggestion
- Be familiar with the BC Gov Github Organization which has guidelines and requirements for hosting government developed open source repositories. 
```

File: README.md [hunk 4]
```diff
@@ -3,6 +3,61 @@
...
+
+Before the course starts, we ask that all attendees install git and GitBash and have a basic text editor such as Wordpad, Notepad++, or SublimeText. Instructions for how to install these tools are found in the [Introduction to git Section](). If anyone is having troubles getting the software up and running, please contact us **before** the course starts so that we can hit the ground running during the workshop. 
+
+### Day 1
+
+ * Section 1. What is version control?
+ * Section 2. Working locally with git
+ * BREAK ☕
+ * Section 3. Working remotely with GitHub
+ * Section 4.
```
Comment (lindsay-fredrick) [inline README.md:59]:
```suggestion
 * Section 4. The BC Gov GitHub Organization
```

File: course-outline.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+- Download Git for Windows and bundled software (includes Git Bash)
+- Create a GitHub account with m2FA, PAT for https connection
+- Configure git
+- Configure RStudio (optional)
+- Pre-workshop assignment to show local configuration and connection to GitHub
+
+Course Reference Material:
+
+- Installation instructions here: `glue::glue(safepaths::get_network_path(), "/8. Data Science Training/github_how_to")`
```
Comment (lindsay-fredrick) [inline course-outline.md:19]: This doesn't work if you aren't using R

File: course-outline.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+Course Reference Material:
+
+- [SW Carpentry Episode 1. Automated Version Control](https://swcarpentry.github.io/git-novice/01-basics.html)
+
+### Section 2. Working local with git
+
+Learning Objective: Use command-line git to track local file changes. (*more about git*) \
+Slides: [Section 2 Slides](https://htmlpreview.github.io/?https://github.com/bcgov/ds-intro-to-git/blob/learning-objectives/slides/02-working-local/02-working-local.html)
+
+Topics:
```
Comment (lindsay-fredrick) [inline course-outline.md:43]: I would be tempted to specifically include the topic of working areas vs staging areas vs the repository.

File: course-outline.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+Course Reference Material:
+
+- [SW Carpentry Episode 6. Ignoring Things](https://swcarpentry.github.io/git-novice/06-ignore.html)
+- [SW Carpentry Episode 7. Remotes in GitHub](https://swcarpentry.github.io/git-novice/07-github.html)
+- [bcgov-r-geo-workshop Lesson 6](https://github.com/bcgov/bcgov-r-geo-workshop/blob/master/20191105_Day_1_PM_Intro/lessons/06-github.md)
+- [Github: Keeping your Account and Data Secure](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github)
+
+### Section 4
+
+Learning Objective: Understand how the bcgov framework, policy and related tools help bc government employees share code on GitHub \
```
Comment (lindsay-fredrick) [inline course-outline.md:85]:
```suggestion
Learning Objective: Understand how the bcgov framework, policy and related tools help bc government employees share code on GitHub \
```
```suggestion
Learning Objective: Understand how the bcgov framework, policy and related tools help B.C. government employees share code on GitHub \
```

File: course-outline.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+Slides: [Section 4 Slides](https://htmlpreview.github.io/?https://github.com/bcgov/ds-intro-to-git/blob/learning-objectives/slides/04-bcgov-org/04-bcgov-org.html)
+
+Topics:
+
+- what is the bcgov organization framework
+- bcgov policy framework and licensing
+- code-sharing and teams
+- introduction to bcgovr
```
Comment (lindsay-fredrick) [inline course-outline.md:93]: I would move this to the end as it's more an R thing than a git thing.

File: course-outline.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+- bcgov-c, bcgov enterprise
+
+Course Reference Material:
+
+- [SW Carpentry Episode 10 - 13](https://swcarpentry.github.io/git-novice/10-open.html)
```
Comment (lindsay-fredrick) [inline course-outline.md:98]:
```suggestion
- [SW Carpentry Episode 10. Open Science](https://swcarpentry.github.io/git-novice/10-open.html)
- [SW Carpentry Episode 11. Licensing](https://swcarpentry.github.io/git-novice/11-licensing.html))
- [SW Carpentry Episode 12. Citation](https://swcarpentry.github.io/git-novice/12-citation.html)
- [SW Carpentry Episode 13. Hosting](https://swcarpentry.github.io/git-novice/13-hosting.html)
```

**Review by lindsay-fredrick — COMMENTED** [review]
Seems like a fine start to me - I just left a couple of minor comments but I'm looking forward to seeing the materials come together!

File: course-outline.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+Topics:
+
+- what is the bcgov organization framework
+- bcgov policy framework and licensing
+- code-sharing and teams
+- introduction to bcgovr
+- bcgov-c, bcgov enterprise
+
+Course Reference Material:
```
Comment (lindsay-fredrick) [inline course-outline.md:96]: I might also include links to useful bcgov materials:

 [Content Approval Checklist](https://github.com/bcgov/BC-Policy-Framework-For-GitHub/blob/master/BC-Open-Source-Development-Employee-Guide/Content-Approval-Checklist.md)

[BC Gov Organizations in GitHub](https://docs.developer.gov.bc.ca/bc-government-organizations-in-github/)

[Policy Framework](https://github.com/bcgov/BC-Policy-Framework-For-GitHub/tree/master)

File: README.md (hunk 1 above)
Comment (stephhazlitt) [inline README.md:8]:
```suggestion
The goal of this workshop is to teach new-to-programming data professionals how to harness the power of version control by giving attendees a foundation in the fundamentals of version control with git.
```

Comment (stephhazlitt) [reply README.md:8]: Maybe add a url to git? e.g. [git](https://git-scm.com)

File: README.md [hunk 11]
```diff
@@ -3,6 +3,61 @@
...
+- Connect to GitHub, create a remote repository and push local changes to GitHub
+- (*Something about BC Gov specific materials Org, Licensing, bcgovr*)
+
+The workshop content will follow best practices for version control such as writing descriptive commit messages, including a .gitignore file and tracking relevent project files.
+
+## Who should take this course?
+
+ * Anyone looking to solidify their foundational understanding of version control
+ * Anyone keen to learn what git or GitHub is (**no prior experience necessary!**)
+ * Anyone wondering how version control can be used to manage and collaborate on software projects
```
Comment (stephhazlitt) [inline README.md:29]:
```suggestion
 * Anyone wondering how version control can be used to manage and collaborate on software and code-based data science projects
```

File: README.md [hunk 12]
```diff
@@ -3,6 +3,61 @@
...
+
+## Previous Offerings
+
+ * November 2023 will be the first offering of this course
```
Comment (stephhazlitt) [inline README.md:33]: I suggest we remove this section. We may only offer this internal branch lesson 1x, I want to manage expectation (this feels a bit like we are building a course for scaling out).

File: README.md [hunk 13]
```diff
@@ -3,6 +3,61 @@
...
 
+This is a repository to house materials for a 1/2 day workshop introducing participants to version control using git and the popular code hosting platform, GitHub.
+
+The goal of this workshop is to teach new-to-programming data professionals how to harness the power of version control by giving attendees a foundation in the fundamentals of version control and git.
+
+GitHub is a widely-used platform for hosting and sharing code repositories, facilitating seamless collaboration among developers.  Git, on the other hand, is a popular version control system that enables you to track changes in your code, collaborate with team members, and maintain a history of your project's development. 
+
+By the end of this course, you will have gained a fundamental understanding of version control, enabling you to:
+
+- Set up a github account and install git (pre-workshop)
```
Comment (stephhazlitt) [inline README.md:14]:
```suggestion
- Set up a GitHub account and install the software git (pre-workshop)
```

File: README.md [hunk 14]
```diff
@@ -3,6 +3,61 @@
...
+- Understand the value-add of using version control to manage software projects
```
Comment (stephhazlitt) [inline README.md:15]:
```suggestion
- Understand the value-add of using version control to manage software and code-based data science projects
```

File: README.md [hunk 15]
```diff
@@ -3,6 +3,61 @@
...
+- Understand the basics of how automated version control systems work.
```
Comment (stephhazlitt) [inline README.md:16]:
```suggestion
- Understand the basics of how automated version control systems work
```

File: README.md [hunk 16]
```diff
@@ -3,6 +3,61 @@
...
+- Use command-line git to track local file changes
```
Comment (stephhazlitt) [inline README.md:17]:
```suggestion
- Use git, popular command-line version control software, to track local file changes
```

File: course-outline.md [hunk 17]
```diff
@@ -0,0 +1,99 @@
+## Course Outline: Introduction to Version Control 
+
+### Pre-Workshop Installation and Set Up
+
+Learning Objective: Set up a github account and install git (pre-workshop). \
```
Comment (stephhazlitt) [inline course-outline.md:5]:
```suggestion
Learning Objective: Set up a GitHub account and install git (pre-workshop). \
```

File: course-outline.md (hunk 5 above)
Comment (stephhazlitt) [reply course-outline.md:19]: I suggest we focus on git only, and leave aside the R parts. Also, is it worth including Mac OS instructions?

**Comment by stephhazlitt** [comment]
Nice work getting the content together @BonnieJRobert. I wonder if it might be cleaner to convert fully to `.quarto` (the repo has a mix of `.rmd` and `.quarto`), and I think there might be a way to have the slides and sections but not duplicate all the overhead files/components (the PR has over 300 files because each page is it's own site I think)? If you want to deploy the learning material as a "site" with quarto, you can use `_quarto.yaml`, see [here for an example](https://quarto.org/docs/websites/).

**Review by stephhazlitt — APPROVED** [review]

File: course-outline.md (hunk 5 above)
Comment (BonnieJRobert) [reply course-outline.md]: @stephhazlitt Do you have experience with the [Git for Mac installer](https://sourceforge.net/projects/git-osx-installer/files/)?  I recall using `homebrew`, but this means another download.

**Review by lindsay-fredrick — APPROVED** [review]

**Pull request merged**